arangod/Aql/OptimizerRules.cpp

```diff
@@ -3109,7 +3111,20 @@ struct SortToIndexNode final
     bool const isOnlyAttributeAccess =
         (!sortCondition.isEmpty() && sortCondition.isOnlyAttributeAccess());
 
-    if (isOnlyAttributeAccess && isSorted && !isSparse && sortCondition.isUnidirectional() &&
+    // FIXME: why not just call index->supportsSortCondition here always?
+    if (index->type() == Index::IndexType::TRI_IDX_TYPE_INVERTED_INDEX) {
+      auto support = index->supportsSortCondition(&sortCondition, outVariable, 1);
+      if (support.supportsCondition) {
+        // sort condition is fully covered by index... now we can remove the
+        // sort node from the plan
+        _plan->unlinkNode(_plan->getNodeById(_sortNode->id()));
+        // we need to have a sorted result later on, so we will need a sorted
+        // GatherNode in the cluster
+        indexNode->needsGatherNodeSort(true);
+        _modified = true;
+        handled = true;
```

Comment:
The code that is executed here for an inverted index that covers the sort condition is completely identical to the code below that is executed if a regular index covers the sort condition.
Can we simplify it to something like:
```
bool indexCoversSortCondition = false;
if (invertex index supports sort condition) {
  indexCoversSortCondition = true;
} else if (regular index supports sort condition) {
  indexCoversSortCondition = true;
}

if (indexCoversSortCondition) {
  // sort condition is fully covered by index... now we can remove the
  // sort node from the plan
  _plan->unlinkNode(_plan->getNodeById(_sortNode->id()));
  // we need to have a sorted result later on, so we will need a sorted
  // GatherNode in the cluster
  indexNode->needsGatherNodeSort(true);
  _modified = true;
  handled = true;
}
```